pkg/service/init.go

```diff
@@ -0,0 +1,95 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package service
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/edgexfoundry/device-sdk-go/internal/autoevent"
+	"github.com/edgexfoundry/device-sdk-go/internal/cache"
+	"github.com/edgexfoundry/device-sdk-go/internal/clients"
+	"github.com/edgexfoundry/device-sdk-go/internal/common"
+	"github.com/edgexfoundry/device-sdk-go/internal/container"
+	"github.com/edgexfoundry/device-sdk-go/internal/controller"
+	"github.com/edgexfoundry/device-sdk-go/internal/provision"
+	dsModels "github.com/edgexfoundry/device-sdk-go/pkg/models"
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/startup"
+	"github.com/edgexfoundry/go-mod-bootstrap/di"
+	"github.com/gorilla/mux"
+)
+
+// Bootstrap contains references to dependencies required by the BootstrapHandler.
+type Bootstrap struct {
+	router *mux.Router
+}
+
+// NewBootstrap is a factory method that returns an initialized Bootstrap receiver struct.
+func NewBootstrap(router *mux.Router) *Bootstrap {
+	return &Bootstrap{
+		router: router,
+	}
+}
+
+func (b *Bootstrap) BootstrapHandler(ctx context.Context, wg *sync.WaitGroup, _ startup.Timer, dic *di.Container) (success bool) {
+	controller.LoadRestRoutes(b.router, dic)
+
+	common.CurrentConfig = container.ConfigurationFrom(dic.Get)
```

Comment:
Please consider creating a new issue to refactor to remove using global variables.

Reply:
I've tried it but couldn't come up with an efficient way to do so. 
How would you suggest do achieve this? Do we just pass the dic container all around in SDK or is there any better solution ?

Reply:
Check out the core services and how they handle this. We spent quite a bit of effort making things like the logging client, registry client, and external service clients lazy loaded and not global.

 It's certainly something that should be addressed in a follow on, because it is quite a bit of work. But, you're right, the answer is to just pass around parameters everywhere in a manner not too dissimilar from how I got the wait group into my code in my PR.

Reply:
ok I'll create a new issue once this PR is merged.